avocado/utils/cloudinit.py

```diff
@@ -150,18 +150,14 @@ def do_POST(self):
 
         Respond with status 200 if the instance phoned back.
         """
-        path = self.path[1:]
-        if path[-1] == "/":
-            path = path[:-1]
+        path = self.path.rstrip("/")
```

Comment:
The previous code also removed the first character, therefore your changes change the behavior and tests are failing. What about to use `self.path[1:].rstrip("/")` ?